storage/client/blob/src/main/java/com/azure/storage/blob/BlockBlobClient.java

```diff
@@ -161,32 +163,42 @@ public BlobOutputStream getBlobOutputStream(BlobAccessConditions accessCondition
 
         Mono<Response<BlockBlobItem>> upload = blockBlobAsyncClient.blockBlobAsyncRawClient
             .upload(fbb.subscribeOn(Schedulers.elastic()), length, headers, metadata, accessConditions)
-            .map(rb -> new SimpleResponse<>(rb, new BlockBlobItem(rb.deserializedHeaders())));;
+            .map(rb -> new SimpleResponse<>(rb, new BlockBlobItem(rb.deserializedHeaders())));
 
         try {
             return Utility.blockWithOptionalTimeout(upload, timeout);
-        }
-        catch (UncheckedIOException e) {
+        } catch (UncheckedIOException e) {
             throw e.getCause();
```

Comment:
I have seen you have the Utility.blockWithOptionalTimeout for simplifying try/catch somewhere else. Why you try/catch again here? Can you move the checking into the utility methods?

Reply:
The blockWithOptionalTimeout simplifies and if/else block not a try/catch block. It shouldn't be used for catching IOExceptions everywhere